### PR TITLE
Ref: Skip CI for commits without code changes

### PR DESCRIPTION
**Pull request by lucas-zimerman**
#3419

## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description

Goal is to make CI faster when a PR only alters files such as changelog, readme,...
This was done with a reusable job that allows different group of jobs to know if they need to run or not.

## :bulb: Motivation and Context

It frees resources to Sentry (mostly the Mac Virtual machines) when small changes are made and also allow tests to pass faster when code-less changes are made.

## :green_heart: How did you test it?
The last PRs have a comment `TEST:...` that checks specific scenarios, for example, only changing the changelog to not trigger the build to run, altering some source file to check if the build and tests will run, breaking the Branch check (worst case) to validate that even if the job fails, the remain jobs will continue working.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes (no permanent tests but tested by simulating commits to test specific cases)
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

### CODE REVIEW

**Comment by lucas-zimerman** [comment]
Current Commit test:
- ignore .yml files temporarily (changes added in order to skip the builds for this PR)
- .prettierignore altered and ignored

File: src/js/sdk.tsx
Comment (krystofwoldrich) [inline src/js/sdk.tsx:1]: We should remove this.
```suggestion
```

File: .github/workflows/skip-ci.yml
Comment (krystofwoldrich) [inline .github/workflows/skip-ci.yml:26]: We only have 5 md files, could we just add them to the skip list to make this more explicit? 

Screenshot 2024-01-31 at 12 19 00

**Comment by krystofwoldrich** [comment]
Possible I've missed some benefits of the job, but could we use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore?

**Comment by lucas-zimerman** [comment]
> Possible I've missed some benefits of the job, but could we use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore?

Yeah we can, the idea was to reuse the list across the jobs, instead of having to place it on each workflow/trigger

we can place the following snippet on each file, but it seems like the build and test jobs are stucked forever, despite falling on Tte path-ignore list ( could be a github bug)
```
    paths-ignore:
      - '.prettierignore'
      - 'CHANGELOG.md'
      - 'README.md'
      - '.github/CODEOWNERS'
      - '.github/pull_request_template.md'
      - '.github/ISSUE_TEMPLATE/BUG_REPORT.md'
      - '.github/ISSUE_TEMPLATE/FEATURE_REQUEST.md'
```

But the main difference is with paths ignore we'll have to place this snippet on each workflow and the end result will look like this
image
Where with the reusable workflow, we'll have to add an additional job to run it but the end result will not hide the jobs but mark as skipped.
image


For the time being, I applied the requested change, too bad github doesn't have more reusable options for this use case.

File: .github/workflows/buildandtest.yml
Comment (krystofwoldrich) [inline .github/workflows/buildandtest.yml:24]:
```suggestion
      - '.github/ISSUE_TEMPLATE/FEATURE_REQUEST.md'

```

**Comment by krystofwoldrich** [comment]
@lucas-zimerman Looks good, thank you for all the tests and screenshots. 

I will check with @kahest the required jobs settings and then we can merge it.

**Review by krystofwoldrich — APPROVED** [review]
I approve this to see if GH allows the merge or if it will be blocked by the required jobs.

**Comment by krystofwoldrich** [comment]
This can be merged after we remove the required Test and Build jobs.

**Comment by krystofwoldrich** [comment]
I reverted the PR back to skip-ci action as it's compatible with the required jobs. The ignore-paths are not working for the repository as they don't allow us to require certain jobs like Test and Build.

**Review by krystofwoldrich — APPROVED** [review]
Looks good! 🚀

**Pull request merged**
